static_assert is only included with two arguments

Since C++17, there is also a [`static_assert` with a single argument](https://en.cppreference.com/w/cpp/language/static_assert).
This is not included in Clair and causes a NullPointerException if we try to parse it using:

```
|lib://clair/src/lang/cpp/M3.rsc|(9983,280,<176,0>,<177,243>): Java(
  "UnsupportedOperationException",
  "Conversion to AST did not work for org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTLiteralExpression at |file:///path/to/file.cpp|(XXX,XX)",
  Java("NullPointerException",""))
        at lang.cpp.internal.Parser.parseCppToM3AndAst(|unknown:///Parser.java|(0,0,<492,0>,<492,0>))
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(|unknown:///NativeMethodAccessorImpl.java|(0,0,<0,0>,<0,0>))
        at parseCppToM3AndAst(|lib://clair/src/lang/cpp/M3.rsc|(10256,5,<177,236>,<177,241>))
        at main(|file:///file/that/executed/parser.rsc|(679,15,<29,4>,<29,19>))
        at $shell$(|prompt:///|(0,30,<1,0>,<1,30>)WARNING: ResourcesPlugin was null, can't get workspace; not overriding include files
ok
```

Executing the same piece of code after having executed `:set errors true` in the same Rascal terminal, we get a proper stack-trace that points us to the missing `static_assert`:
```
java.lang.UnsupportedOperationException: Conversion to AST did not work for org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTLiteralExpression at |file:///path/to/file.cpp|(XXX,XX)
        at lang.cpp.internal.Parser.parseCppToM3AndAst(Parser.java:492)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.rascalmpl.interpreter.result.JavaMethod.invoke(JavaMethod.java:241)
        at org.rascalmpl.interpreter.result.JavaMethod.call(JavaMethod.java:163)
        at org.rascalmpl.semantics.dynamic.Expression$CallOrTree.interpret(Expression.java:540)
        at org.rascalmpl.semantics.dynamic.Statement$Expression.interpret(Statement.java:365)
        at org.rascalmpl.semantics.dynamic.Statement$Assignment.interpret(Statement.java:205)
        at org.rascalmpl.interpreter.result.RascalFunction.runBody(RascalFunction.java:385)
        at org.rascalmpl.interpreter.result.RascalFunction.call(RascalFunction.java:326)
        at org.rascalmpl.semantics.dynamic.Expression$CallOrTree.interpret(Expression.java:540)
        at org.rascalmpl.semantics.dynamic.Statement$Expression.interpret(Statement.java:365)
        at org.rascalmpl.interpreter.result.RascalFunction.runBody(RascalFunction.java:385)
        at org.rascalmpl.interpreter.result.RascalFunction.call(RascalFunction.java:294)
        at org.rascalmpl.semantics.dynamic.Expression$CallOrTree.interpret(Expression.java:540)
        at org.rascalmpl.semantics.dynamic.Command$Expression.interpret(Command.java:61)
        at org.rascalmpl.interpreter.Evaluator.eval(Evaluator.java:1094)
        at org.rascalmpl.interpreter.Evaluator.eval(Evaluator.java:955)
        at org.rascalmpl.interpreter.Evaluator.eval(Evaluator.java:908)
        at org.rascalmpl.repl.RascalInterpreterREPL.evalStatement(RascalInterpreterREPL.java:132)
        at org.rascalmpl.repl.BaseRascalREPL.handleInput(BaseRascalREPL.java:106)
        at org.rascalmpl.vscode.lsp.terminal.LSPTerminalREPL$1.handleInput(LSPTerminalREPL.java:191)
        at org.rascalmpl.repl.BaseREPL.handleInput(BaseREPL.java:180)
        at org.rascalmpl.repl.BaseREPL.run(BaseREPL.java:347)
        at org.rascalmpl.vscode.lsp.terminal.LSPTerminalREPL.main(LSPTerminalREPL.java:269)
Caused by: java.lang.NullPointerException
        **at lang.cpp.internal.Parser.visit(Parser.java:1097)**
        at lang.cpp.internal.Parser.visit(Parser.java:986)
        at org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTStaticAssertionDeclaration.accept(CPPASTStaticAssertionDeclaration.java:78)
        at lang.cpp.internal.Parser.visit(Parser.java:3712)
        at lang.cpp.internal.Parser.visit(Parser.java:3546)
        at org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTDeclarationStatement.accept(CPPASTDeclarationStatement.java:72)
        at lang.cpp.internal.Parser.lambda$visit$47(Parser.java:3691)
        at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
        at lang.cpp.internal.Parser.visit(Parser.java:3690)
        at lang.cpp.internal.Parser.visit(Parser.java:3542)
        at org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTCompoundStatement.accept(CPPASTCompoundStatement.java:88)
        at lang.cpp.internal.Parser.visit(Parser.java:1228)
        at lang.cpp.internal.Parser.visit(Parser.java:965)
        at org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTFunctionDefinition.accept(CPPASTFunctionDefinition.java:191)
        at lang.cpp.internal.Parser.lambda$visit$35(Parser.java:1961)
        at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
        at lang.cpp.internal.Parser.visit(Parser.java:1960)
        at lang.cpp.internal.Parser.visit(Parser.java:1893)
        at lang.cpp.internal.Parser.visit(Parser.java:1868)
        at org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTCompositeTypeSpecifier.accept(CPPASTCompositeTypeSpecifier.java:174)
        at lang.cpp.internal.Parser.visit(Parser.java:1298)
        at lang.cpp.internal.Parser.visit(Parser.java:968)
        at org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTSimpleDeclaration.accept(CPPASTSimpleDeclaration.java:100)
        at lang.cpp.internal.Parser.visit(Parser.java:1113)
        at lang.cpp.internal.Parser.visit(Parser.java:992)
        at org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTTemplateDeclaration.accept(CPPASTTemplateDeclaration.java:115)
        at lang.cpp.internal.Parser.lambda$visit$48(Parser.java:4067)
        at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
        at lang.cpp.internal.Parser.visit(Parser.java:4066)
        at org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTNamespaceDefinition.accept(CPPASTNamespaceDefinition.java:123)
        at lang.cpp.internal.Parser.visit(Parser.java:836)
        at org.eclipse.cdt.internal.core.dom.parser.ASTTranslationUnit.accept(ASTTranslationUnit.java:278)
        at lang.cpp.internal.Parser.convertCdtToRascal(Parser.java:628)
        at lang.cpp.internal.Parser.parseCppToM3AndAst(Parser.java:476)
        ... 26 more
)
|lib://clair/src/lang/cpp/M3.rsc|(9983,280,<176,0>,<177,243>): Java(
  "UnsupportedOperationException",
  "Conversion to AST did not work for org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTLiteralExpression at |file:///path/to/file.cpp|(XXX,XX)",
  Java("NullPointerException",""))
        at lang.cpp.internal.Parser.parseCppToM3AndAst(|unknown:///Parser.java|(0,0,<492,0>,<492,0>))
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(|unknown:///NativeMethodAccessorImpl.java|(0,0,<0,0>,<0,0>))
        at parseCppToM3AndAst(|lib://clair/src/lang/cpp/M3.rsc|(10256,5,<177,236>,<177,241>))
        at main(|file:///file/that/executed/parser.rsc|(679,15,<29,4>,<29,19>))
        at $shell$(|prompt:///|(12,4,<1,12>,<1,16>)WARNING: ResourcesPlugin was null, can't get workspace; not overriding include files
```
(relevant line highlighted)

--- a/src/lang/cpp/AST.rsc
+++ b/src/lang/cpp/AST.rsc
@@ -116,6 +116,7 @@ data Declaration(list[Attribute] attributes = [], loc src=|unknown:///|, bool is
     | \alias(Name \alias, Expression mappingTypeId, loc decl = |unknown:///|)
     
     | \staticAssert(Expression condition, Expression message) //no attributes
+    | \staticAssert(Expression condition) //no attributes
     
     | \explicitTemplateInstantiation(Declaration declaration) //no attributes
     | \explicitTemplateInstantiation(Modifier modifier, Declaration declaration) //no attributes

--- a/src/lang/cpp/internal/AST.java
+++ b/src/lang/cpp/internal/AST.java
@@ -198,6 +198,8 @@ public class AST {
     = tf.constructor(typestore,_Declaration,"defaultedFunctionDefinition",_DeclSpecifier,"declSpecifier",tf.listType(_Expression),"memberInitializer",_Declarator,"declarator");
   private static final Type _Declaration_functionDefinition_4 
     = tf.constructor(typestore,_Declaration,"functionDefinition",_DeclSpecifier,"declSpecifier",_Declarator,"declarator",tf.listType(_Expression),"memberInitializer",_Statement,"body");
+  private static final Type _Declaration_staticAssert_1 
+    = tf.constructor(typestore,_Declaration,"staticAssert",_Expression,"condition");
   private static final Type _Declaration_baseSpecifier_1 
     = tf.constructor(typestore,_Declaration,"baseSpecifier",tf.listType(_Modifier),"modifiers");
   private static final Type _Declaration_parameter_1 
@@ -2533,6 +2535,21 @@ public class AST {
     
     if ($isMacroExpansion) kwParams.put("isMacroExpansion", vf.bool(true));
     return vf.constructor(_Declaration_functionDefinition_4 , $declSpecifier, $declarator, $memberInitializer, $body).asWithKeywordParameters().setParameters(kwParams);
+  }
+  
+  public IConstructor Declaration_staticAssert(IConstructor $condition, ISourceLocation $loc, boolean $isMacroExpansion) {
+      
+    if (!$condition.getType().isSubtypeOf(_Expression)) {
+      throw new IllegalArgumentException("Expected " + _Expression + " but got " + $condition.getType() + " for $condition:" + $condition);
+    }
+    
+    Map<String, IValue> kwParams = new HashMap<String, IValue>();
+    
+    kwParams.put("src", $loc);
+    
+    
+    if ($isMacroExpansion) kwParams.put("isMacroExpansion", vf.bool(true));
+    return vf.constructor(_Declaration_staticAssert_1 , $condition).asWithKeywordParameters().setParameters(kwParams);
   }
   
   public IConstructor Declaration_baseSpecifier(IList $modifiers, ISourceLocation $loc, ISourceLocation $decl, boolean $isMacroExpansion) {

--- a/src/lang/cpp/internal/Parser.java
+++ b/src/lang/cpp/internal/Parser.java
@@ -1094,8 +1094,14 @@ public class Parser extends ASTVisitor {
 		boolean isMacroExpansion = isMacroExpansion(declaration);
 		declaration.getCondition().accept(this);
 		IConstructor condition = stack.pop();
-		declaration.getMessage().accept(this);
-		stack.push(builder.Declaration_staticAssert(condition, stack.pop(), loc, isMacroExpansion));
+		ICPPASTLiteralExpression msg = declaration.getMessage();
+		if (msg != null) {
+			msg.accept(this);
+			stack.push(builder.Declaration_staticAssert(condition, stack.pop(), loc, isMacroExpansion));
+		}
+		else {
+			stack.push(builder.Declaration_staticAssert(condition, loc, isMacroExpansion));
+		}
 		return PROCESS_ABORT;
 	}
 

